src/app/shared/media/component/media.component.ts

```diff
@@ -33,15 +54,132 @@ export class MediaComponent {
     map(mimeType => mimeType.startsWith('video'))
   );
 
-  constructor(private readonly sanitizer: DomSanitizer) {}
+  readonly playWithAndroidNativePlayer$ = combineLatest([
+    this.isVideo$,
+    of(Capacitor.getPlatform() == 'android'),
+  ]).pipe(map(([isVideo, isAndroid]) => isVideo && isAndroid));
+
+  @Input()
+  set filePath(value: string | undefined) {
+    if (value) this.previewVideoFilePath$.next(value);
+  }
+
+  @ViewChild('previewVideo', { static: false })
+  previewVideoRef: ElementRef | null = null;
+
+  constructor(
+    private readonly sanitizer: DomSanitizer,
+    private readonly ref: ChangeDetectorRef,
+    private readonly renderer: Renderer2,
+    private readonly elementRef: ElementRef
+  ) {}
+
+  ngAfterViewInit(): void {
+    const waitForWebviewToRecover = 700;
+    this.createNativePlayerTimeoutId = window.setTimeout(() => {
+      this.createNativePlayer();
+    }, waitForWebviewToRecover);
+  }
 
-  // eslint-disable-next-line class-methods-use-this
   onImageLoaded() {
     this.isLoading$.next(false);
   }
 
-  // eslint-disable-next-line class-methods-use-this
   onVideoLoaded() {
     this.isLoading$.next(false);
   }
+
+  /**
+   * WORKAROUND: https://github.com/numbersprotocol/capture-lite/issues/2845
+   * Purpose: Use android native player to play 4k videos.
+   * 4k videos cannot be played with HTML video tag on some Android devices because
+   * of Android's WebView limitation. Using a native player helps to circumvent this issue.
+   * Note: iOS WebViews can play 4k videos without any issues.
+   */
+  private createNativePlayer() {
+    this.playWithAndroidNativePlayer$
+      .pipe(
+        filter(playWithAndroidNativePlayer => playWithAndroidNativePlayer),
+        switchMap(() => this.previewVideoFilePath$),
+        tap(_ => {
+          this.onVideoLoaded();
+          this.ref.detectChanges();
+        }),
+        switchMap(previewVideoFilePath => {
+          if (this.previewVideoRef?.nativeElement) {
+            return PreviewVideo.create({
+              id: this.previewVideoId,
+              src: previewVideoFilePath,
+              element: this.previewVideoRef.nativeElement,
+              config: {},
+            });
+          }
+          throw Error('previewVideoRef not found.');
+        }),
+        map(previewVideo => {
+          this.addCSSClass();
+          return (this.previewVideo = previewVideo);
+        }),
+        catchError((_err: unknown) => {
+          /**
+           * Error handling: In case the creation of the native player fails,
+           * an error should be reported to the crash reporting system (currently not in place).
+           * TODO: Implement error reporting to the crash reporting system when available.
+           */
+          // Error not reported due to absence of crash reporting system.
```

Comment:
How about show an error message to let user knows the native player fails instead of silently fails?
Although we can't collect error stacktrace in the absence of crash reporting system, it might help if user can know an error happened and report their device & Android version.

Reply:
@shc261392 , sure I will add user friendly error message.

Reply:
@shc261392 done [fix(media.component): show user friendly error message](https://github.com/numbersprotocol/capture-lite/pull/2868/commits/af3f01746720d045bcf5544543c8d7f6975b121d)